kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/MapEq.kt

```diff
@@ -33,26 +41,28 @@ private fun generateError(actual: Map<*, *>?, expected: Map<*, *>?): Throwable {
 }
 
 private fun buildFailureMessage(actual: Map<*, *>?, expected: Map<*, *>?): String {
-   val keysDifferentAt = when {
+   return when {
       actual != null && expected != null -> {
          val keysHavingDifferentValues = actual.keys.filterNot { expected[it] == actual[it] }
-         "Values differed at keys ${keysHavingDifferentValues.joinToString(limit = 10)}"
+         "Values differed at keys ${keysHavingDifferentValues.joinToString(limit = 10)}\n"
       }
+
       else -> ""
    }
-   return "Expected\n${expected?.toFormattedString()}\nto be equal to\n${actual?.toFormattedString()}\n$keysDifferentAt"
 }
 
 
-private fun Map<*, *>.toFormattedString(): String {
-   if (isEmpty()) return "{}"
+private fun Map<*, *>?.print(): Printed {
+   if (this == null) return null.show()
+   if (isEmpty()) return Printed("{}")
    val indentation = "  "
    val newLine = "\n"
-   return toList()
+   return Printed(toList()
       .joinToString(
          separator = ",$newLine",
          prefix = "{$newLine",
          postfix = "$newLine}",
          limit = 10
       ) { "$indentation${it.first.show().value} = ${it.second.show().value}" }
+   )
```

Comment:
Would it make more sense to just use `actual.show()` ? I saw there's an implementation of `MapShow` which I tried to use, but I felt this formatting was more readable.

Reply:
We don't have to use show, if we feel this looks better.